packages/firestore/src/util/async_queue.ts

```diff
@@ -39,24 +40,25 @@ export class AsyncQueue {
    * Adds a new operation to the queue. Returns a promise that will be resolved
    * when the promise returned by the new operation is (with its value).
    *
-   * Can optionally specify a delay to wait before queuing the operation.
+   * Can optionally specify a delay (in milliseconds) to wait before queuing the
+   * operation.
    */
   schedule<T>(op: () => Promise<T>, delay?: number): Promise<T> {
     if (this.failure) {
       fail('AsyncQueue is already failed: ' + this.failure.message);
     }
 
     if ((delay || 0) > 0) {
-      this.delayedOpCount++;
       const deferred = new Deferred<T>();
-      setTimeout(() => {
+      let handle = window.setTimeout(() => {
```

Comment:
Why `window.setTimeout()` instead of the global? This is incompatible with node without extra hacks to define window.

Reply:
`setTimeout()` is defined to (perhaps incorrectly) return a Timer. See https://github.com/Microsoft/TypeScript/issues/842#issuecomment-58213196